### DISCUSSION TITLE
Best way to use from legacy app (.net 4.5.2)

### DISCUSSION THREAD

**Discussion by lofti198** [Tyrrrz/YoutubeExplode#772] [Q&A] [answered]
Installed old version 4.7.16.0 , since need to use from old legacy .net framework app. It has different classes structure and I am trying to pick the working code. however there is a question - will it even work, since I assume Youtube could change output data structure since that time?

**Comment by lofti198** [comment]
Getting such extension right away:
![image](https://github.com/Tyrrrz/YoutubeExplode/assets/42960488/8dafbfaa-d4ac-438c-b7e7-b3d03883d11f)

{"Thread was being aborted."}

**Comment by Tyrrrz** [comment, marked as answer]
The short answer is no, it won't work on legacy .NET 4.5.2. You should upgrade at least to 4.6.1 or, better yet, to .NET 5+:

![image](https://github.com/Tyrrrz/YoutubeExplode/assets/1935960/bc127a7a-8b57-4efb-9317-2fe5b2c2c46b)

**Comment by Sekoree** [comment]
As a last resort you could create a Net7/8 AOT library and get what you need through FFI exports, super inefficient but works I guess, I do that for a Unity thing that is stuck on Net3.5

**Reply by lofti198** [reply]
wow cool idea, however I ve managed to switch my app to 4.7.2 and it worked perfectly!!
